CAIPs/caip-25.md

```diff
@@ -25,11 +25,28 @@ The motivation comes from the lack of standardization across blockchains to expo
 
 ## Specification
 
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" written in
+uppercase in this document are to be interpreted as described in [RFC
+2119][]
+
+### Definition
+
 The session is proposed by a caller and the response by the respondent is used as the baseline for an ongoing session that both parties will persist.
-The properties and authorization scopes that make up the session are expected to be persisted and tracked over time by both parties in a discrete data store, identified by an entropic [identifier][CAIP-171] assigned in the initial response.
-This object gets updated, extended, closed, etc. by successive calls and notifications, each tagged by this identifier.
+- When a wallet responds with a success response containing a `sessionId` (an entropic [identifier][CAIP-171]), the properties and authorization scopes that make up the session are expected to be persisted and tracked over the life of the session by both parties in a discrete data store.
+- When the wallet does not provide a `sessionId` in its initial response, the wallet MUST persist and track the properties and authorization scopes that make up the session. In this case, the wallet MUST implement a method [`wallet_getSession`][CAIP-286] to enable the caller to query for the current status of the session at any time.
+
+After a session is established between wallet and caller, subsequent `wallet_createSession` calls are used to update the properties and authorization scopes of the session.
+- When a `sessionId` is returned in the initial `wallet_createSession` response, subsequent `wallet_createSession` calls MUST either contain the previously used `sessionId` (on the root of the request) in which case that session is modified, or the caller must generate a new `sessionId` and pass it with the request, in which case a new session is created and the previous session dangles in parallel (until its expiration if applicable) though maintaining concurrent sessions is discouraged (see Security Considerations).
```

Comment:
A note that I need to add definition for corner cases when say a caller calls `wallet_createSession` with an unknown `sessionId` since this is a new case as `sessionId` was never a parameter to `wallet_createSession` previously

Reply:
The parameter is added to the spec [here](https://github.com/ChainAgnostic/CAIPs/pull/285/files#diff-0864406dba09fa71f65989c7b3d779c71ee5cfc4cc39dec39c33c3eb96c8d35eR126)